extensions/data-workspace/src/services/workspaceService.ts

```diff
@@ -158,6 +159,56 @@ export class WorkspaceService implements IWorkspaceService {
 		return vscode.workspace.workspaceFile ? this.getWorkspaceConfigurationValue<string[]>(ProjectsConfigurationName).map(project => this.toUri(project)) : [];
 	}
 
+	/**
+	 * Check for projects that are in the workspace folders but have not been added to the workspace through the dialog or by editing the .code-workspace file
+	 */
+	async checkForProjectsNotAddedToWorkspace(): Promise<void> {
+		const config = vscode.workspace.getConfiguration(constants.projectsConfigurationKey);
+
+		// only check if the user hasn't selected not to show this prompt again
+		if (config[constants.showNotAddedProjectsMessageKey]) {
+			// get the unique supported project extensions
+			const supportedProjectExtensions = [...new Set((await this.getAllProjectTypes()).map(p => { return p.projectFileExtension; }))];
+
+			// look for any projects that haven't been added to the workspace
+			const projectsInWorkspace = this.getProjectsInWorkspace();
+			const workspaceFolders = vscode.workspace.workspaceFolders;
+
+			if (workspaceFolders) {
+				for (const extType of supportedProjectExtensions) {
+					// find all project files with extType in each workspace folder
+					for (const folder of workspaceFolders) {
+						// path needs to use forward slashes for glob to work
+						const escapedPath = glob.escapePath(folder.uri.fsPath.replace(/\\/g, '/'));
+						const projFilter = path.posix.join(escapedPath, '**', `*.${extType}`);
+						const results = await glob(projFilter);
+
+						let containsNotAddedProject = false;
+						for (const projFile of results) {
+							// if any of the found projects aren't already in the workspace's projects, we can stop checking and show the info message
+							if (!projectsInWorkspace.find(p => p.fsPath === projFile)) {
+								containsNotAddedProject = true;
+								break;
+							}
+						}
+
+						if (containsNotAddedProject) {
```

Comment:
did you put this in the wrong place, if this is added here, user will be prompted multiple times.

Reply:
this should be done only once, right?

Reply:
there's a return at the end of this if block so the notification only shows once since the function gets exited already. I put this here so that the function will get finished as soon as possible because the other workspace folders don't need to be checked if a not added project was already found.